src/node/settings.ts

```diff
@@ -0,0 +1,77 @@
+const os = require('os');
+const fsPromises = require('fs/promises');
+const fs = require('fs');
+const path = require('path');
+
+import type { Pin } from '../types';
+import { macDest, windowsDest } from './constants';
+import { encode } from './encoding';
+import { get } from './request';
+import type { Settings } from './types';
+
+const device = os.type();
+
+async function getSettings(): Promise<Settings> {
+  let settingsFile: string;
+
+  if (device === 'Windows_NT') {
+    settingsFile = await retrieveSettingsFile(windowsDest);
+    const settingsIni: Settings = JSON.parse(settingsFile);
+    return settingsIni;
+  }
+
+  settingsFile = await retrieveSettingsFile(macDest);
+  const settingsIni: Settings = JSON.parse(settingsFile);
+  return settingsIni;
+}
+
+async function setSettingsApiKey(apiKey: string, pin: Pin | undefined) {
+  try {
+    await get('/api/v2/integrations/appmixer/user-profile', apiKey);
+  } catch (error) {
+    throw new Error('Invalid API key');
+  }
+
+  const settings = await getSettings();
+  const newSettings = { ...settings, apiKey, hasPin: false };
+
+  if (pin) {
+    const secret = `${pin.first}${pin.second}${pin.third}${pin.fourth}`;
```

Comment:
can pin be a class that can return the pin and avoid doing this everywhere?

Reply:
not done yet